### PR TITLE
Dyt.sim

### PR DESCRIPTION
**Pull request by duanyutong**
### Background

cannot call the actual ImpedanceControlCallback() function because in sim there's no Robot instance, no Model instance needed for impedance calculations

### What's new
- ✅ [New feature description. Only a single new major feature is allowed per PR.]
- ❌ New feature is accompanied by new test: [name and description of new test].

### Related work
- ❌ This PR needs [link]
- ❌ [link] needs this PR

### TODOs / Nice-To-Haves
- ❌ [Add new unit test for new feature.]
- ❌ [Add new system test for new feature.]

### Tests
1. ✅ Tested on simulated robot: [Describe the tests/commands used.]
2. ❌ Tested on physical robot: [Describe the tests/commands used.]

### Quality
3. ✅ New code is written in pure C++17 to the best of my knowledge.
4. ✅ New code follows established C++17 best practices ([C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines)).
5. ✅ New code passed `clang-format` and `cpplint` at least, if not all of our static code analysers.
6. ✅ I have included Doxygen-style documentation in the new C++ code.

### Note to reviewers
Please verify that all sections are accurately filled out and that all 6 numbered entries above are present and ticked/checked off, with their requirements met, if applicable.

### CODE REVIEW

File: src/driver/franka_plan_runner.cc [hunk 1]
```diff
@@ -420,45 +420,77 @@ int FrankaPlanRunner::RunSim() {
...
+      if (!callback) {  // first time here
+        dexai::log()->info("Starting sim impedance control...");
+        SetCompliantPushParameters(robot_state, Eigen::Vector3d(0, 0, 0.050));
+        comm_interface_->SetCompliantPushActive(true);
+        comm_interface_->ClearCompliantPushStartRequest();
+        callback = 2;
+      }
+      // keep pushing until stop requested
+      // cannot call the actual ImpedanceControlCallback() function
+      // as model_->coriolis() segfaults in sim
```
Comment (5yler) [inline src/driver/franka_plan_runner.cc:471]: That's because `model_` is nullptr

File: src/driver/franka_plan_runner.cc [hunk 2]
```diff
@@ -863,10 +894,8 @@ franka::JointPositions FrankaPlanRunner::JointPositionCallback(
       return franka::MotionFinished(output_to_franka);
     }
 
-    {
-      // make a copy and pass by value
-      const auto plan_utime {plan_utime_};
-      auto overtime_warning {[plan_utime]() {
+    {  // threaded logging, capture member vars by val
+      auto overtime_warning {[plan_utime = plan_utime_]() {
```
Comment (5yler) [inline src/driver/franka_plan_runner.cc:898]: Where is this actually threaded?

Comment (5yler) [reply src/driver/franka_plan_runner.cc:898]: Oh nvm didn't see if with collapsed diff

**Review by 5yler — APPROVED** [review]
LGTM platform & style

File: src/driver/franka_plan_runner.cc (hunk 2 above)
Comment (duanyutong) [reply src/driver/franka_plan_runner.cc:898]: yep L836 `std::thread overtime_warning_thread {overtime_warning};`

**Pull request merged**